### PR TITLE
test: Moved KFP backend visualization tests to GitHub Actions

### PR DESCRIPTION
**Pull request by hbelmiro**
Resolves: https://github.com/kubeflow/pipelines/issues/10985

PR to remove the tests from GoogleCloudPlatform/oss-test-infra:
- https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2316

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: .github/workflows/backend-visualization.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    branches: [ master ]
+
+  pull_request:
+    paths:
+      - '.github/workflows/backend-visualization.yml'
+      - 'backend/src/apiserver/visualization/**'
+      - 'test/presubmit-backend-visualization.sh'
+
+jobs:
+  samples-v2-test:
```
Comment (DharmitD) [inline .github/workflows/backend-visualization.yml:14]: aren't `backend visualization` and `samples-v2` two different tests?

https://github.com/GoogleCloudPlatform/oss-test-infra/blob/3738b70f78b971cc281b1791cf8e73cfbe08ae55/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L179

https://github.com/GoogleCloudPlatform/oss-test-infra/blob/3738b70f78b971cc281b1791cf8e73cfbe08ae55/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L189

Comment (hbelmiro) [reply .github/workflows/backend-visualization.yml]: Yes. I forgot to rename the job.
Fixed.

File: .github/workflows/backend-visualization.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: KFP backend visualization tests
```
Comment (DharmitD) [inline .github/workflows/backend-visualization.yml]: would suggest changing it back to `KFP Backend Visualization Tests`, to keep the `KFP` suffix name convention consistent across all workflow names. For instance: https://github.com/kubeflow/pipelines/blob/master/.github/workflows/presubmit-backend.yml#L1

Also to be able to differentiate between KFP and kfp-tekton workflows: https://github.com/kubeflow/pipelines/blob/master/.github/workflows/backend.yml#L1

**Comment by DharmitD** [comment]
@hbelmiro could you squash commits to a single commit?

File: .github/workflows/backend-visualization.yml (hunk 2 above)
Comment (hbelmiro) [reply .github/workflows/backend-visualization.yml]: Makes sense.
Done.

**Review by DharmitD — COMMENTED** [review]
/lgtm

**Pull request merged**